utils/gyb_syntax_support/AttributeNodes.py

```diff
@@ -197,5 +199,122 @@
+         children=[
+             Child('LeftParen', kind='LeftParenToken'),
+             Child('DiffParams', kind='DifferentiationParamList',
+                   collection_element_name='DifferentiationParam',
+                   description='The parameters for differentiation.'),
+             Child('RightParen', kind='RightParenToken'),
+         ]),
+
+    # differentiation-param-list ->
+    #     differentiation-param differentiation-param-list?
+    Node('DifferentiationParamList', kind='SyntaxCollection',
+         element='DifferentiationParam'),
+
+    # differentiation-param -> ('self' | identifer) ','?
+    Node('DifferentiationParam', kind='Syntax',
+         description='''
+         A differentiation parameter: either the "self" identifier or a \
+         function parameter name.
+         ''',
+         traits=['WithTrailingComma'],
+         children=[
+             Child('Parameter', kind='Syntax',
+                   node_choices=[
+                       Child('Self', kind='SelfToken'),
+                       Child('Name', kind='IdentifierToken'),
+                   ]),
+             Child('TrailingComma', kind='CommaToken', is_optional=True),
+         ]),
+
+    # differentiable-attr-func-specifier ->
+    #     ('jvp' | 'vjp') ':' func-decl-name ','?
+    Node('DifferentiableAttributeFuncSpecifier', kind='Syntax',
+         description='''
+         A function specifier, consisting of an identifier, colon, and a \
+         function declaration name (e.g. `vjp: foo(_:_:)`).
+         ''',
+         traits=['WithTrailingComma'],
+         children=[
+             Child('Label', kind='IdentifierToken',
+                   text_choices=['jvp', 'vjp']),
+             Child('Colon', kind='ColonToken'),
+             Child('FunctionDeclName', kind='FunctionDeclName',
+                   description='The referenced function name.'),
+             Child('TrailingComma', kind='CommaToken', is_optional=True),
+         ]),
+
+    # func-decl-name -> (identifier | operator) decl-name-arguments?
+    # NOTE: This is duplicated with `DeclName` above. Change `DeclName`
+    # description and use it if possible.
+    Node('FunctionDeclName', kind='Syntax',
+         description='A function declaration name (e.g. `foo(_:_:)`).',
+         children=[
+             Child('Name', kind='Syntax', description='''
+                   The base name of the referenced function.
+                   ''',
+                   node_choices=[
+                       Child('Identifier', kind='IdentifierToken'),
+                       Child('PrefixOperator', kind='PrefixOperatorToken'),
+                       Child('SpacedBinaryOperator',
+                             kind='SpacedBinaryOperatorToken'),
+                   ]),
+             Child('Arguments', kind='DeclNameArguments',
+                   is_optional=True, description='''
+                   The argument labels of the referenced function, optionally \
```

Comment:
This description string should be fixed. Currently, it looks like this in [`Sources/SwiftSyntax/gyb_generated/syntax_nodes/SyntaxNodes.swift`]( https://github.com/apple/swift-syntax/pull/178/files#diff-be6fd49b9da56aa4d28b732c3f377cd8R7571):

```swift
  /// 
  /// The argument labels of the referenced function, optionally                    specified.
  /// 
  public var arguments: DeclNameArgumentsSyntax? {
```
A corresponding change should be made to [apple/swift-syntax](https://github.com/apple/swift-syntax) to update gyb-generated sources, e.g. https://github.com/apple/swift-syntax/pull/178.